src/WordSearch/WordSearchSolver.ts

```diff
@@ -208,4 +221,46 @@ export class WordSearchSolver {
     }
     return true;
   }
+
+  private checkSpaces(): void {
+    if (
+      this._matrix.length === 0 ||
+      this._spaceTreatment === WordSearchSpaceTreatment.None
+    ) {
+      return;
+    } else if (this._spaceTreatment === WordSearchSpaceTreatment.RemoveAll) {
```

Comment:
This entire condition seems like it should be merged with the next one. The behavior of the two is the same when `inPuzzle = true` so we can set the default value of `inPuzzle` based on the setting the caller chose. Then the rest of the algorithm runs as written.